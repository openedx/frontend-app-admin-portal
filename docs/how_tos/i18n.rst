####################
React App i18n HOWTO
####################

This document has moved to the frontend-platform repo: https://github.com/openedx/frontend-platform/blob/master/docs/how_tos/i18n.rst
