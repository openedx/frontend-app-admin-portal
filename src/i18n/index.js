import { messages as paragonMessages } from '@edx/paragon';

export default [
  paragonMessages,
];
