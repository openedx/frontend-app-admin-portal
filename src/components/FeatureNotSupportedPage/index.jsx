import React from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from '@edx/frontend-platform/i18n';
import {
  Alert,
} from '@openedx/paragon';
import {
  Info,
} from '@openedx/paragon/icons';

export const FeatureNotSupported = () => (
  <>
    <Helmet>
      <title>Feature not supported</title>
    </Helmet>
    <div className="text-center py-5">
      <Alert variant="info" icon={Info}>
        <FormattedMessage
          id="admin.portal.feature.not.supported.page.message"
          defaultMessage="This feature is currently unavailable in this environment."
          description="The message displayed on the feature not supported page"
        />
      </Alert>
    </div>
  </>
);

const FeatureNotSupportedPage = () => (
  <main role="main" style={{ flex: 1 }}>
    <div className="container-fluid mt-3">
      <FeatureNotSupported />
    </div>
  </main>
);

export default FeatureNotSupportedPage;
