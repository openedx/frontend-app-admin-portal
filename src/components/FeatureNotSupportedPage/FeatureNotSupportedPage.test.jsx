import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import FeatureNotSupportedPage from './index';

describe('<FeatureNotSupportedPage />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <FeatureNotSupportedPage />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
