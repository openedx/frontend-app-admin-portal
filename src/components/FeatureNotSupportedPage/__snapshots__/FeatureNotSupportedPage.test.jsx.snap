// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureNotSupportedPage /> renders correctly 1`] = `
<main
  role="main"
  style={
    {
      "flex": 1,
    }
  }
>
  <div
    className="container-fluid mt-3"
  >
    <div
      className="text-center py-5"
    >
      <div
        className="fade alert-content alert alert-info show"
        role="alert"
      >
        <span
          className="pgn__icon alert-icon"
        >
          <svg
            aria-hidden={true}
            fill="none"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm1 15h-2v-6h2v6Zm0-8h-2V7h2v2Z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          className="pgn__alert-message-wrapper"
        >
          <div
            className="alert-message-content"
          >
            This feature is currently unavailable in this environment.
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
